aws/resource_aws_cloudwatch_event_bus_policy.go

```diff
@@ -0,0 +1,179 @@
+	}
+
+	d.SetId(eventBusName)
+
+	return resourceAwsCloudWatchEventBusPolicyRead(d, meta)
+}
+
+// See also: https://docs.aws.amazon.com/AmazonCloudWatchEvents/latest/APIReference/API_DescribeEventBus.html
+func resourceAwsCloudWatchEventBusPolicyRead(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).cloudwatcheventsconn
+
+	eventBusName := d.Id()
+
+	input := events.DescribeEventBusInput{
+		Name: aws.String(eventBusName),
+	}
+	var output *events.DescribeEventBusOutput
+	var err error
+	var policy *string
+
+	// Especially with concurrent PutPermission calls there can be a slight delay
+	err = resource.Retry(iamwaiter.PropagationTimeout, func() *resource.RetryError {
+		log.Printf("[DEBUG] Reading CloudWatch Events bus: %s", input)
+		output, err = conn.DescribeEventBus(&input)
+		if err != nil {
+			return resource.NonRetryableError(fmt.Errorf("reading CloudWatch Events permission (%s) failed: %w", d.Id(), err))
+		}
+
+		policy, err = getEventBusPolicy(output)
+		if err != nil {
+			return resource.RetryableError(err)
+		}
+		return nil
+	})
+
+	if isResourceTimeoutError(err) {
+		output, err = conn.DescribeEventBus(&input)
+		if output != nil {
+			policy, err = getEventBusPolicy(output)
+		}
+	}
+
+	if isResourceNotFoundError(err) {
+		log.Printf("[WARN] Policy on {%s} EventBus not found, removing from state", d.Id())
+		d.SetId("")
+		return nil
+	}
+	if err != nil {
+		return fmt.Errorf("error reading policy from CloudWatch EventBus (%s): %w", d.Id(), err)
+	}
+
+	busName := aws.StringValue(output.Name)
+	if busName == "" {
+		busName = tfevents.DefaultEventBusName
+	}
+	d.Set("event_bus_name", busName)
+
+	d.Set("policy", policy)
+
+	return nil
+}
+
+func getEventBusPolicy(output *events.DescribeEventBusOutput) (*string, error) {
+	if output == nil || output.Policy == nil {
+		return nil, &resource.NotFoundError{
+			Message:      fmt.Sprintf("Policy for CloudWatch EventBus %s not found", *output.Name),
+			LastResponse: output,
+		}
+	}
+
+	return output.Policy, nil
+}
+
+func resourceAwsCloudWatchEventBusPolicyUpdate(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).cloudwatcheventsconn
+
+	eventBusName := d.Id()
+
+	input := events.PutPermissionInput{
+		EventBusName: aws.String(eventBusName),
+		Policy:       aws.String(d.Get("policy").(string)),
+	}
+
+	log.Printf("[DEBUG] Update CloudWatch EventBus policy: %s", input)
+	_, err := conn.PutPermission(&input)
+	if isAWSErr(err, events.ErrCodeResourceNotFoundException, "") {
+		log.Printf("[WARN] CloudWatch EventBus %q not found, removing from state", d.Id())
+		d.SetId("")
+		return nil
+	}
+	if err != nil {
+		return fmt.Errorf("error updating policy for CloudWatch EventBus (%s): %w", d.Id(), err)
+	}
+
+	return resourceAwsCloudWatchEventBusPolicyRead(d, meta)
+}
+
+func resourceAwsCloudWatchEventBusPolicyDelete(d *schema.ResourceData, meta interface{}) error {
```

Comment:
Could you create a test for this? e.g. https://github.com/hashicorp/terraform-provider-aws/blob/7247dd06a08d9647ac061d387e5a185ae948efe6/aws/resource_aws_cloudwatch_event_permission_test.go#L294